pkg/collector/corechecks/servicediscovery/module/impl_linux.go

```diff
@@ -6,24 +6,52 @@
 package module
 
 import (
+	"fmt"
+	"net"
 	"net/http"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/procfs"
+	"github.com/shirou/gopsutil/v3/process"
+	"github.com/vishvananda/netns"
 
 	"github.com/DataDog/datadog-agent/cmd/system-probe/api/module"
 	sysconfigtypes "github.com/DataDog/datadog-agent/cmd/system-probe/config/types"
+	"github.com/DataDog/datadog-agent/cmd/system-probe/utils"
 	"github.com/DataDog/datadog-agent/comp/core/telemetry"
 	workloadmeta "github.com/DataDog/datadog-agent/comp/core/workloadmeta/def"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/servicediscovery/model"
+	"github.com/DataDog/datadog-agent/pkg/util/kernel"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
 	"github.com/DataDog/datadog-agent/pkg/util/optional"
 )
 
+const (
+	patchListeners = "/listeners"
+)
+
 // Ensure discovery implements the module.Module interface.
 var _ module.Module = &discovery{}
 
+// cacheData holds process data that should be cached between calls to the
+// endpoint.
+type cacheData struct {
+	env []string
+}
```

Comment:
Where and How do you ensure the cache cannot bombard the memory of system-probe?

Reply:
Now it's only caching the service name. It's only for running, listening processes so it's proportional to that but I'll create a JIRA to check if we need to limit it in any other way as discussed.